src/Umbraco.Core/DisposableObject.cs

```diff
@@ -2,60 +2,58 @@
 using System.Threading;
 
 namespace Umbraco.Core
-{
-	/// <summary>
-	/// Abstract implementation of IDisposable.
-	/// </summary>
-	/// <remarks>
-	/// Can also be used as a pattern for when inheriting is not possible.
-	/// 
+{
+    /// <summary>
+    /// Abstract implementation of IDisposable.
+    /// </summary>
+    /// <remarks>
+    /// Can also be used as a pattern for when inheriting is not possible.
+    /// 
     /// See also: https://msdn.microsoft.com/en-us/library/b1yfkh5e%28v=vs.110%29.aspx
     /// See also: https://lostechies.com/chrispatterson/2012/11/29/idisposable-done-right/
     /// 
     /// Note: if an object's ctor throws, it will never be disposed, and so if that ctor
     /// has allocated disposable objects, it should take care of disposing them.
-	/// </remarks>
-	public abstract class DisposableObject : IDisposable
-	{
-		private bool _disposed;
-	    private readonly object _locko = new object();
-
-		// gets a value indicating whether this instance is disposed.
-        // for internal tests only (not thread safe)
-        //TODO make this internal + rename "Disposed" when we can break compatibility
-		public bool IsDisposed { get { return _disposed; } }

```

Comment:
@RobertCopilau you can't change the original class :P That will be a breaking change, what I meant was change the new class of `DisposableObjectSlim`, since it's new we can make it however we want :)

Reply:
Oops, alright I'll fix that :)